R/guides-.R

```diff
@@ -485,15 +485,76 @@ Guides <- ggproto(
       return(zeroGrob())
     }
 
+    # extract the guide position
+    positions <- vapply(
+      params,
+      function(p) p$position[1] %||% default_position,
+      character(1), USE.NAMES = FALSE
+    )
+
     # Populate key sizes
     theme$legend.key.width  <- calc_element("legend.key.width",  theme)
     theme$legend.key.height <- calc_element("legend.key.height", theme)
 
-    grobs <- self$draw(theme, default_position, theme$legend.direction)
+    grobs <- self$draw(theme, positions, theme$legend.direction)
+    keep <- !vapply(grobs, is.zero, logical(1), USE.NAMES = FALSE)
+    grobs <- grobs[keep]
     if (length(grobs) < 1) {
       return(zeroGrob())
     }
-    grobs <- grobs[order(names(grobs))]
+
+    # prepare the position of inside legends
+    default_inside_just <- valid.just(
+      calc_element("legend.justification.inside", theme)
+    )
+    default_inside_position <- calc_element(
+      "legend.position.inside", theme
+    )
+    inside_justs <- inside_positions <- vector("list", length(positions))
+
+    # we grouped the legends by the positions, for inside legends, they'll be
+    # splitted by the actual inside coordinate
+    groups <- positions
+    for (i in seq_along(positions)) {
+      if (identical(positions[i], "inside")) {
```

Comment:
```suggestion
    for (i in seq_along(positions)[positions == "inside"]) {
```

Reply:
Here, in this loop, I tried to keep the the length of the `inside_justs` variable so that we can use `keep` variable to subset this.

Reply:
That will continue to be the case with the proposed change.
If you find it easier to read, we can also use:

```r
for (i in which(positions == "inside")) {
```

Reply:
Sorry for this miss